packages/shorebird_code_push_client/lib/src/code_push_client.dart

```diff
@@ -37,14 +39,19 @@ class CodePushClient {
   CodePushClient({
     http.Client? httpClient,
     Uri? hostedUri,
+    Logger? logger,
```

Comment:
It might be too much effort at this stage but one thought I had was instead of injecting a logger and having to modify `CodePushClient`, we could have created a `LoggingHttpClient` which extends `http.Client` and injected that into `CodePushClient` to keep `code_push_client` decoupled from logging/mason_logger/meta, etc.

Reply:
Done